brain/section.h

```diff
@@ -0,0 +1,135 @@
+/* Copyright (c) 2013-2015, EPFL/Blue Brain Project
+ *                          Juan Hernando <jhernando@fi.upm.es>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef BRAIN_SECTION
+#define BRAIN_SECTION
+
+#include <brain/api.h>
+#include <brain/types.h>
+
+namespace brain
+{
+
+namespace detail
+{
+class Morphology;
+}
+
+/**
+ * A class to represent a morphological section.
+ *
+ * A Section is an unbranched piece of a morphological skeleton.
+ * This class provides functions to query information about the sample points
+ * that conform the section and functions to obtain the parent and children
+ * sections.
+ *
+ * The cell soma is also considered a section, but some functions have
+ * special meaning for it.
+ *
+ * Sections cannot be directly created, but are returned by several
+ * brain::Morphological and brain::Section methods.
+ *
+ * This is a lightweight object with STL container style thread safety.
+ * It is also safe to use a section after the morphology from where it comes
+ * has been deallocated. The morphological data will be keept as long as there
+ * is a Section referring to it.
+ */
+class Section
+{
+public:
+    friend class Morphology;
+
+    Section( const Section& section );
+
+    ~Section();
+
+    Section& operator=( const Section& section );
+
+    bool operator==( const Section& section ) const;
+
+    /** Return the ID of this section. */
+    uint32_t getID() const;
+
+    SectionType getType() const;
+
+    /**
+     * Return the total length of this section.
+     *
+     * If this section is a soma section the length is ill-defined and this
+     * function will return 0.
+     */
+    BRAIN_API float getLength() const;
+
+    /**
+     * Return the list of all point samples that define this section.
+     *
+     * If this sections is a soma section return the list of points in the
+     * soma ring.
```

Comment:
Good questions... and I don't have a good answer that's the reason for which is soma profile poly-line is returned. A user that needs an empty vector won't call this method. A user that needs the center could use 0, 0, 0 when the morphology it's known to be centered there, or compute the centroid otherwise.
I can describe what the soma ring is supposed to be.